src/Bicep.Core/Emit/ExpressionConverter.cs

```diff
@@ -35,8 +35,8 @@ public LanguageExpression ConvertExpression(SyntaxBase expression)
                 case BooleanLiteralSyntax boolSyntax:
                     return CreateFunction(boolSyntax.Value ? "true" : "false");
 
-                case NumericLiteralSyntax numericSyntax:
-                    return new JTokenExpression(numericSyntax.Value);
+                case IntegerLiteralSyntax integerSyntax:
+                    return integerSyntax.Value > int.MaxValue || integerSyntax.Value < int.MinValue ? CreateFunction("json", new JTokenExpression(integerSyntax.Value.ToString())) : new JTokenExpression((int)integerSyntax.Value);
```

Comment:
>ToString [](start = 166, length = 8)

ToString() defaults to current culture according to docs at [https://docs.microsoft.com/en-us/dotnet/api/system.int64.tostring?view=net-5.0#System_Int64_ToString](https://docs.microsoft.com/en-us/dotnet/api/system.int64.tostring?view=net-5.0#System_Int64_ToString). Can you explicitly pass InvariantCulture here?

Reply:
I used `ToInvariantString`, is that ok?

Reply:
Yup, that works too.